logging/impl_test.go

```diff
@@ -607,3 +607,38 @@ func TestLoggingDeduplication(t *testing.T) {
 	assertLogMatches(t, notStdout,
 		`2023-10-30T13:19:45.806Z	INFO	impl	logging/impl_test.go:132	foo	{"key":"value"}`)
 }
+
+func TestNeverDeduplicate(t *testing.T) {
+	// Create a logger object that will write to the `notStdout` buffer. Explicitly
+	// set `DeduplicateLogs` to true on the registry for the logger.
+	registry := newRegistry()
+	registry.DeduplicateLogs.Store(true)
+
+	// Artificially lower noisy message window for testing.
+	originalNoisyMessageWindowDuration := noisyMessageWindowDuration
+	noisyMessageWindowDuration = 500 * time.Millisecond
+	defer func() {
+		noisyMessageWindowDuration = originalNoisyMessageWindowDuration
+	}()
+
+	// Assert that `NeverDeduplicate` method causes loggers to not deduplicate.
+	notStdout := &bytes.Buffer{}
+	logger := &impl{
+		name:                     "impl",
+		level:                    NewAtomicLevelAt(DEBUG),
+		appenders:                []Appender{NewWriterAppender(notStdout)},
+		registry:                 registry,
+		testHelper:               func() {},
+		recentMessageCounts:      make(map[string]int),
+		recentMessageEntries:     make(map[string]LogEntry),
+		recentMessageWindowStart: time.Now(),
+	}
+	logger.NeverDeduplicate()
+
+	// Log 4 identical messages, and assert that all four appear.
+	for range 4 {
```

Comment:
wait what?

Reply:
I did not know about this. If I wanted to assign the current loop value to some variable `x`, is there a short form for that? Or do I have to write `for x := 0; x < 4; x++ {`?

Reply:
Yeah! It's [a new feature of Go 1.22](https://tip.golang.org/doc/go1.22#introduction). You can also assign to `x` with `for x := range`.